updated psc.tf - fixed workspace psc bug

syedjavedahmed opened the pull request:
Updated psc.tf to fix a bug related to workspace PSC endpoints for both backend_psc_ep and spoke_frontend_psc_ep. The previous logic was fetching an incorrect target, causing the PSC resources to remain in a pending status. 

Previous Logic:
`target = "projects/prod-gcp-${var.google_region}/regions/${var.google_region}/serviceAttachments/plproxy-psc-endpoint-all-ports"` 
This resulted in incorrect target values, leading to pending PSC resource status.

Updated logic:
 `target = "projects/general-prod-${replace(var.google_region,"-","")}-01/regions/${var.google_region}/serviceAttachments/plproxy-psc-endpoint-all-ports"`

This update aligns with the correct format and retrieves the correct PSC service target, exemplified by: 
`projects/general-prod-asiasoutheast1-01/regions/asia-southeast1/serviceAttachments/plproxy-psc-endpoint-all-ports`

Changes:
Implemented replace(var.google_region, "-", "") function to correctly format the region value.

micheledaddetta-databricks commented:
@alexott, I reviewed the issue. Project IDs related to PSC for each region do not always follow this pattern. The best option, IMO, is to use a map that contains the proper project ID for each region. I can implement it in the next few days.

syedjavedahmed commented:
Hello @alexott @micheledaddetta-databricks,

I have updated the code to utilize a region-based map that correctly fetches the ProjectId based on the GCP region. I have validated this by provisioning it across multiple regions in my GCP project, and it works accurately as expected.

Pls have a review now, should be good. I have adhered to DBX coding standards here, including comments for clarity. Also, I have squashed my previous commits to create a single commit for this PR.

P.S. We're implementing a similar Soln. here @ Epsilon with GCP - DBX, and I am working on automation using Tf scripts. I thought I could contribute here as well! :)

Lmk if this is good to go!
